### PR TITLE
Add production configuration

### LINKED ISSUE #40: Provision production Prow cluster
The goal of this issue is to prepare official Prow cluster, that is configured for github.com/kyma-project/kyma.
Provision such cluster should be fully automatic. 
Make sure that:
- ingress is using https
Depends on #25

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added production configuration

Example jobs execution on `kyma` repository: https://github.com/kyma-project/kyma/pull/1434

**Related issue(s)**

See #40

### CODE REVIEW

File: prow/cluster/tls-ing_ingerss.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: extensions/v1beta1
```
Comment (aszecowka) [inline prow/cluster/tls-ing_ingerss.yaml]: in `install-prow.sh` script we deploy another ingress from `starter.yaml`. Is it a problem?

File: prow/cluster/cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# Source: cert-manager/templates/00-namespace.yaml
```
Comment (aszecowka) [inline prow/cluster/cert-manager.yaml]: Add readme file explaining whats and when

File: prow/cluster/cert-manager.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: certificates.certmanager.k8s.io
+  annotations:
+    "helm.sh/hook": crd-install
+  labels:
+    app: cert-manager
+    chart: cert-manager-v0.5.0
+    release: cert-manager
+    heritage: Tiller
```
Comment (aszecowka) [inline prow/cluster/cert-manager.yaml]: have you copied it from somewhere? Should we add link to the original sources?

File: prow/plugins.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+plugins:
```
Comment (aszecowka) [inline prow/plugins.yaml]: plugins.yaml and plugins.yaml.tpl should be kept in sync IMO. The same applies for config.yaml

Comment (michal-hudy) [reply prow/plugins.yaml]: Done

File: prow/install-prow.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (aszecowka) [inline prow/install-prow.sh]: please add documentation how this file differs from another install-prow.sh

File: prow/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+The installation script performs the following steps to install Prow:
+
+- Deploy the NGINX Ingress Controller.
+- Create a ClusterRoleBinding.
+- Deploy Prow components with the `a202e595a33ac92ab503f913f2d710efabd3de21`revision.
+- Deploy Cert Manager.
+- Deploy Secure Ingress.
+- Removes Insecure Ingress
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- Removes Insecure Ingress.
```

File: prow/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+Install the following tools:
+
+- Kubernetes 1.10+ on GKE
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [gcloud](https://cloud.google.com/sdk/gcloud/)
+
+### Google Cloud Platform configuration
+
+- Global static IP Address with name `prow-production`
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- A global static IP address with the `prow-production` name.
```

File: prow/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+- DNS registry for domain `status.build.kyma-project.io` that points to `prow-production` address
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- A DNS registry for the `status.build.kyma-project.io` domain that points to the `prow-production` address.
```

File: prow/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+### Secrets:
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
### Secrets
```

File: prow/README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
+# Prow Production Configuration
+
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
## Overview
This project contains a set of configuration files for the Prow production cluster.
```

File: prow/README.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+## Prerequisites
+
+### Tools
```
Comment (kazydek) [inline prow/README.md]:
```suggestion

```

File: prow/README.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+Install the following tools:
+
+- Kubernetes 1.10+ on GKE
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [gcloud](https://cloud.google.com/sdk/gcloud/)
+
+### Google Cloud Platform configuration
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
### Google Cloud Platform configuration

These are the requirements for the GCP configuration to provision Prow on the production cluster:
```

File: prow/README.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+- Global static IP Address with name `prow-production`
+- DNS registry for domain `status.build.kyma-project.io` that points to `prow-production` address
+
+### Secrets:
+
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
The obligatory Secrets include:
```

File: prow/README.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+- hmac-token - Prow HMAC token used for GitHub Webhooks
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- `hmac-token` which is a Prow HMAC token used for GitHub Webhooks.
```

File: prow/README.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+- oauth-token - GitHub token with read/write access to `kyma-bot` account
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- `oauth-token` which is a GitHub token with read and write access to the `kyma-bot` account.
```

File: prow/README.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+- compute-service-account - Google Cloud Service Account with roles:
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
- `compute-service-account` which is a Google Cloud service account with the following roles:
```

File: prow/README.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+- Deploy Secure Ingress.
+- Removes Insecure Ingress
+
+To check if the installation is successful, perform the following steps:
+
+1. Check if all Pods are up and running:
+   `kubeclt get pods`
+2. Check if the Deck is accessible from outside of the cluster:
+   `kubectl get ingress tls-ing`
+   Copy the address of the ingress `tls-ing` and open it in a browser to display the Prow status on the dashboard.
```
Comment (kazydek) [inline prow/README.md]:
```suggestion
   Copy the address of the `tls-ing` Ingress and open it in a browser to display the Prow status on the dashboard.
```

**Pull request merged**